app/services/issue_notification.rb

```diff
@@ -0,0 +1,27 @@
+class IssueNotification < ApplicationService
+  def initialize(task:)
+    @task = task
+  end
+
+  def call
+    build_notification_object.notify(task)
+  end
+
+  private 
+
+  attr_reader :task
+
+  NOTIFICATION_DECORATORS = {
+    'sms'      => SmsNotificationDecorator,
+    'whatsapp' => WhatsappNotificationDecorator,
+    'email'     => MailNotificationDecorator,
+    'telegram' => TelegramNotificationDecorator
+  }.freeze
+
+  def build_notification_object
+    NOTIFICATION_DECORATORS
+      .select { |notification, notification_decorator| task.user.notification_preferences['task_completed'].include?(notification) }
+      .values
+      .reduce(TaskNotificationDecorator.new) { |notification, notification_class| notification_class.new(notification) }
```

Comment:
```suggestion
    notifications = task.user.notification_preferences['task_completed']
    notifications.each { |notification| NOTIFICATION_DECORATORS[notification].new(notification) }
```

I think you can reduce the complexity of this method using this. If you're using `select` with the `NOTIFICATION_DECORATORS` to match the notifications withing `task.user.notification_preferences['task_completed']` you can do the opposite and selecet the decorator directly with the hash key. Let me know if this works.